include/rpc/detail/make_unique.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#ifndef MAKE_UNIQUE_H_FOOBAR
+#define MAKE_UNIQUE_H_FOOBAR
+
+#include <memory>
+
+namespace rpc {
+namespace detail {
+
+using std::make_unique;
```

Comment:
When I finalize C++11 support, this will be conditionally either `std::make_unique` or something hand-rolled.

I tried putting the handrolled version into the std namespace, but that can fail in certain cases if the compiler finds an implementation of `make_unique` even if it's not used in the current standard.